src/main.rs

```diff
@@ -367,19 +355,33 @@ fn main() {
     }
     #[cfg(windows)]
     {
+        use future::Either;
+        use futures::future::IntoFuture;
+
         let ctrl_c = tokio_signal::ctrl_c().flatten_stream().into_future();
         let crash = future::poll_fn(move || graceful_crash.poll());
 
-        let interruption = rt
-            .block_on(ctrl_c.select2(crash))
-            .map_err(|_| ())
-            .expect("Neither stream errors");
-
-        use futures::future::Either;
+        let mut ctrl_c_select = ctrl_c.select2(crash);
```

Comment:
@a-rodin do you mind taking a quick peek at this section of the change?

Reply:
@a-rodin `block_on` was changed with `wait` to get this to compile. There are some differences, but I am not sure if they matter, so do check it.

Reply:
If we have access to the runtime we should use `block_on` here, I believe in the past it was required for windows since it needed part of the tokio runtime to execute the signal stream.

Reply:
That's unfortunate, then I'll use some Rc<RefCell> to satisfy the lifetime checks for now, and we should revisit this once #1383 is done, or it will be needed to be done with it.